homeassistant/components/sensor/envirophat.py

```diff
@@ -53,6 +54,14 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Sense HAT sensor platform."""
+    # Detect platform by checking for the onboard sensors showing up on i2c
+    if (subprocess.getoutput("i2cdetect -q -y 1 | grep -o 1d") != '1d' and
```

Comment:
@balloob Sure, will rewrite. The crazy hack of physicall sensing the device is the same as in the current `light.piglow` module that does the same test / setup: https://github.com/home-assistant/home-assistant/blob/70ad06d71c4826b00b54b23090238477a354e776/homeassistant/components/light/piglow.py#L35-L37